prov/util/src/util_attr.c

```diff
@@ -721,11 +721,53 @@ int ofi_check_tx_attr(const struct fi_provider *prov,
 	return 0;
 }
 
-int ofi_check_info(const struct util_prov *util_prov, uint32_t api_version,
-		   const struct fi_info *user_info)
+/* if there are multiple fi_info in the provider */
+int ofi_prov_check_info(const struct util_prov *util_prov,
+			uint32_t api_version,
+			const struct fi_info *user_info,
+			struct fi_info **ret_info)
 {
 	const struct fi_info *prov_info = util_prov->info;
 	const struct fi_provider *prov = util_prov->prov;
+	struct fi_info *info = NULL;
+	int ret, first = 1, res = -FI_ENODATA;
+
+	for ( ; prov_info; prov_info = prov_info->next) {
+		ret = ofi_check_info(util_prov, prov_info,
+				     api_version, user_info);
+	    	if (ret)
+			continue;
+		if (ret_info) {
+			info = fi_dupinfo(prov_info);
+			if (first) {
```

Comment:
Thanks, Arun. Probably, it was initially proposed by Sean, but I didn't get it. Sorry:( I'll rework the function.